Core/clim/input-editing-drei.lisp

```diff
@@ -164,31 +166,62 @@ activated with GESTURE"))
 ;;; rely on internal features and implement input-editor support in
 ;;; CLIM-INTERNALS.
 
+;;; When yanking history of commands on the stream, Following helper function
+;;; ensures the sublists are not evaluated by quoting them. They are
+;;; not meant to be evaluated anyway. (See comment on "define-command" in
+;;; Core/clim-core/commands.lisp).
+
+(defun %quote-sublists (list)
+  (let ((result nil))
+    (loop for item in list
+	  do (if (listp item)
+		 (if (eql (car item) 'quote)
+		     (push item result)
+		     (push `(quote ,item) result))
+		 (push item result)))
+    (reverse result)))
+
 (defun history-yank-next (stream input-buffer gesture numeric-argument)
   (declare (ignore input-buffer gesture numeric-argument))
   (let* ((accepting-type *active-history-type*)
+	 (quoted-sublists nil)
          (history (and accepting-type
                        (presentation-type-history accepting-type))))
     (when history
       (multiple-value-bind (object type)
           (presentation-history-next history accepting-type)
         (when type
-          (presentation-replace-input stream object type (stream-default-view stream)
-                                      :allow-other-keys t
-                                      :accept-result nil))))))
+	  (when (and (equal type 'command-or-form) (listp object))
+	    (when (alexandria:starts-with-subseq "COM-" (symbol-name `,(car object)))
+	      (setf quoted-sublists (%quote-sublists object))))
+	  (when (null quoted-sublists)
+	    (setf quoted-sublists object))
+	  (let ((*print-case* :downcase))
+	    (presentation-replace-input stream quoted-sublists type
+					(stream-default-view stream)
+					:allow-other-keys t
+					:accept-result nil)))))))
 
 (defun history-yank-previous (stream input-buffer gesture numeric-argument)
```

Comment:
Both functions look pretty similar, could they be factored into one function with a parameter next/prev ?

Reply:
We could, but they are Commands and have there own keyboard shortcuts.

Reply:
we could make drei more aware about gesture name aware, define gesture names for yank-next and yank-prev and do if on the gesture (now it would be obscure `(equal gesture '((#\n :meta)))`, taking into account numerical argument (so we go 10 entries up for instance) wouldn't be a bad thing either.

That said I think that would be way out of scope of this PR. So let's leave this duplication be for now.